azurerm/data_source_postgresql_server.go

```diff
@@ -0,0 +1,87 @@
+package azurerm
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourcePostgreSqlServer() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmPostgreSqlServerRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"location": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"fqdn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"administrator_login": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tags.SchemaDataSource(),
+		},
+	}
+}
+
+func dataSourceArmPostgreSqlServerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Postgres.ServersClient
+	ctx, cancel := timeouts.ForRead(meta.(*ArmClient).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("PostgreSql Server %q was not found in Resource Group %q", name, resourceGroup)
+		}
+
+		return fmt.Errorf("Error retrieving PostgreSql Server %q (Resource Group %q): %s", name, resourceGroup, err)
+	}
+
+	if id := resp.ID; id != nil {
+		d.SetId(*resp.ID)
+	}
```

Comment:
we need to set this in all cases; so could we switch this for:

```suggestion
	if resp.ID == nil {
		return fmt.Errorf("Error retrieving PostgreSql Server %q (Resource Group %q): `id` was nil", name, resourceGroup)
	}
	
	d.SetId(*resp.ID)
```